passthrough/usecase/passthrough_usecase.go

```diff
@@ -0,0 +1,140 @@
+package usecase
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	staking "github.com/cosmos/cosmos-sdk/x/staking/types"
+	concentratedLiquidity "github.com/osmosis-labs/osmosis/v25/x/concentrated-liquidity/client/queryproto"
+	lockup "github.com/osmosis-labs/osmosis/v25/x/lockup/types"
+
+	"github.com/osmosis-labs/sqs/domain/mvc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type passthroughUseCase struct {
+	poolsUseCase mvc.PoolsUsecase
+
+	bankQueryClient banktypes.QueryClient
+	stakingQueryClient staking.QueryClient
+	lockupQueryClient lockup.QueryClient
+	concentratedLiquidityQueryClient concentratedLiquidity.QueryClient
+}
+
+var _ mvc.PassthroughUsecase = &passthroughUseCase{}
+
+// NewPassThroughUsecase Creates a passthrough use case
+func NewPassThroughUsecase(grpcURI string, puc mvc.PoolsUsecase) (mvc.PassthroughUsecase, error){
+	grpcClient, err := grpc.Dial(grpcURI, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return nil, err
+	}
+	
+	return &passthroughUseCase{
+		poolsUseCase: puc,
+
+		bankQueryClient: banktypes.NewQueryClient(grpcClient),
+		stakingQueryClient: staking.NewQueryClient(grpcClient),
+		lockupQueryClient: lockup.NewQueryClient(grpcClient),
+		concentratedLiquidityQueryClient: concentratedLiquidity.NewQueryClient(grpcClient),
+	}, nil
+}
+
+func (p *passthroughUseCase) GetAccountCoinsTotal(ctx context.Context, address string) (sdk.Coins, error) {
+	coins := sdk.NewCoins();
+ 	
+  // Bank balances including GAMM shares
+	allBalancesRes, err := p.bankQueryClient.AllBalances(ctx, &banktypes.QueryAllBalancesRequest{
+		Address: address,
+	})
+	if err != nil {
+		return nil, err
+	}
+	
+	for _, balance := range allBalancesRes.Balances {
+		if strings.HasPrefix(balance.Denom, "gamm") {
+			// calc underlying coins from gamm shares
+			splitDenom := strings.Split(balance.Denom, "/")
+			poolID := splitDenom[len(splitDenom)-1]
+			poolIDInt, err := strconv.ParseInt(poolID, 10, 64)
+			if err != nil {
+				return nil, err
+			}
+
+			exitCoins, err := p.poolsUseCase.CalcExitCFMMPool(uint64(poolIDInt), balance.Amount)
+			if err != nil {
+				return nil, err
+			}
+			coins = coins.Add(exitCoins...)
+		} else {
+			coins = coins.Add(balance)
+		}
+	}
```

Comment:
This is whats going to be slow in both backend / the current edge function. (And the current costly thing in frontend, especially for accounts with slippage)

IMO should just ship without tallying balancer shares immediately, and let this come over time. Have this be in a second endpoint perhaps that can come async

Reply:
(If we wanted to ship portfolio w/ edge function only first, until this comes in the backend with proper testing)